### DISCUSSION TITLE
Crazyfile firmware version mismatch

### DISCUSSION THREAD

**Discussion by maichongju** [bitcraze/discussions#284] [Q&A] [answered]
I found out some interesting behaviour of the firmware version number. Previously I fork 2022.05 firmware and compile it into a bin file with the version number `xxxx1`. Then I update the fork with the new 2022.09 version. Then I recompile it. After the recompile, the firmware number showing on the console is `xxxx2`. However, in the bin file itself, the version number still shows as `xxxx1`.

If I clone my fork again to another directory and compile it. The bin file will show the right version `xxxx2`. 

I have compared the two bin files, everything is the same except the version number. 

I don't think it is a big issue, but it is something that I have been observing.

**Comment by krichardsson** [comment, marked as answer]
The version number that is added to the build is generated by the [versionTemplate.py](https://github.com/bitcraze/crazyflie-firmware/blob/master/tools/make/versionTemplate.py) script. It is an old "hack" that uses git to try to figure out what the state of the repo is. 
I don't think anyone considered forks when this was written and I'm not sure what to expect :-) It is also possible that there is a bug, as mentioned it is a bit of a hack.

Contributions are always welcome if you are interested in improving the script :-)
